pkg/providers/aws/provider_redis_metrics.go

```diff
@@ -154,6 +150,33 @@ func (r *RedisMetricsProvider) scrapeRedisCloudWatchMetricData(ctx context.Conte
 }
 
 func buildRedisMetricDataQuery(resourceID string) []*cloudwatch.MetricDataQuery {
-	//todo build this out
-	return nil
+	var metricDataQueries []*cloudwatch.MetricDataQuery
+	for _, metricType := range redisCloudWatchMetricTypes {
+		// Metrics are returned per node for ElastiCache
+		// Poll through the two nodes for ElastiCache ( would like a better way to do this as this is not robust)
```

Comment:
You could use [DescribeReplicationGroups](https://docs.aws.amazon.com/sdk-for-go/api/service/elasticache/#ElastiCache.DescribeReplicationGroups) which would give you [ReplicationGroup](https://docs.aws.amazon.com/sdk-for-go/api/service/elasticache/#DescribeReplicationGroupsOutput) from the output where you can get a node group with the two nodes which will contain the id. T'would save you building them and you could use those for the loop instead of the for loop you have there?

Reply:
Thanks for the suggestion . The Dimension is `CacheClusterId` but that is returned by DescribeReplicationGroup https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_DescribeReplicationGroups.html so better to get it directly.

```bash
 aws cloudwatch list-metrics --region=eu-west-1 | jq -c '.Metrics[]' |grep DatabaseMemoryUsagePercentage
{"Namespace":"AWS/ElastiCache","MetricName":"DatabaseMemoryUsagePercentage","Dimensions":[{"Name":"CacheClusterId","Value":"mwcolabbyoc8zbc5redhatrhmioperatort-5q6x-001"}]}
{"Namespace":"AWS/ElastiCache","MetricName":"DatabaseMemoryUsagePercentage","Dimensions":[{"Name":"CacheClusterId","Value":"mwcolabbyoc8zbc5redhatrhmioperatort-5q6x-002"}]}
```